File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import com.pinyougou.result.ResultPage;

import java.util.List;
import java.util.Map;

public interface BrandService {

     ResultPage<TbBrand> showPageList(int page,int size,TbBrand brand);

    void insert(TbBrand brand);
    void update(TbBrand brand);
    TbBrand findById(Long id);
    void delete(Long[] ids);
    List<Map> brandListForTypeTemplate();
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {
    //调用后台接口
    this.reloadList = function (page, size, model) {
        return $http.post("../brand/showPageList.do?page=" + page + "&size=" + size, model);
    };

    this.findById = function (id) {
        return $http.get("../brand/findById.do?id=" + id);
    }

    this.update = function (brand) {
        return $http.post("../brand/update.do", brand);
    }

    this.insert = function (brand) {
        return $http.post("../brand/insert.do", brand);
    }

    this.delete = function (ids) {
        return $http.get("../brand/delete.do?ids=" + ids);
    }

    this.getBrandList = function () {
        return $http.get('../brand/brandListForTypeTemplate.do');
    }
})